### DISCUSSION TITLE
Dashboard keep showing loading for a long time when import data with wrong format

### DISCUSSION THREAD

**Discussion by hieuwu** [supabase/supabase#16948] [Contribute to Supabase]
### Discussed in https://github.com/orgs/supabase/discussions/13528

<div type='discussions-op-text'>

<sup>Originally posted by **hieuwu** April  8, 2023</sup>
# ⚙️ Summary
Currently, when I import data from CSV file, I notice that if we set the value of a column on Supabase is float4 and the data of this column is a string, the platform will stop working without any response.
For example, my column is `price` with type is `float4`. I import a data set with a row containing value at price column is "this is a string". The platform keep showing loading message forever

<img width="720" alt="Screen Shot 2023-04-08 at 14 33 58" src="https://user-images.githubusercontent.com/43868345/230711338-f1a51f0c-4460-426d-a938-7d775b6bcea0.png">

# 💡 Suggestion
It would be nice if we have some mechanism to verify the format of data for number before performing any operation.

I imported the file below
[product.csv](https://github.com/supabase/supabase/files/11182812/product.csv)
</div>

**Comment by vozmi** [comment]
Hello, @hieuwu! 
I've tried to reproduce your error with simplier data but haven't get infinite loader. 
Instead, I got an error: `Failed to import data: Failed to run sql query: invalid input syntax for type real: "..."`.

I created table with three columns:
- id (int8)
- name (text)
- price (float4)

and imported this [test.csv](https://github.com/supabase/supabase/files/11618617/test.csv) file:
| id | name | price |
| -- | --------- | ------- |
| 1 | apple | 0.99 |
| 2 | banana | 1.55 |
| 3 | coconut | 4.55 |
| 4 | tomato | 2.35 |
| 5 | cucumber | 0.99 |
| 6 | potato | test string |
| 7 | pepper | 1.99 |

Error that I got during import is:
![supabase__test__import-csv](https://github.com/supabase/supabase/assets/70607217/a7586a8a-78a5-473a-bd27-54903157b038)

----

Your csv file is semi-colon separated. So when I tried to import your data I gor another error:
![supabase__test__import-csv-1](https://github.com/supabase/supabase/assets/70607217/72c91899-d32b-4b6c-96f7-76169054b363)

Also found out, that one row in your csv  has more columns than needed
```
TcF68MpGmIGiq5LOvF7u; //productId
Chili, Local Grass-Fed Beef; //name
Chili doesn't get much meatier than this; //description
 this is the kind you dream about for game days and crisp winter nights.; //price
5.8; //image
https://firebasestorage.googleapis.com/v0/b/groceries-store-ad0eb.appspot.com/o/Vegetables%2Fcsm_peppers-plp-desktop_a9c6971df1.jpg?alt=media&token=f986caa0-d65c-4c76-8186-ff75abd6db1e; //category
5igbz6Y8hEfluyhUy3Zp; //nutrition
Chili peppers (Capsicum annuum) are the fruits of Capsicum pepper plants, notable for their hot flavor.  They are members of the nightshade family, related to bell peppers and tomatoes. Many varieties of chili peppers exist, such as cayenne and jalapeño.  Chili peppers are primarily used as a spice and can be cooked or dried and powdered. Powdered, red chili peppers are known as paprika. // !no column!
```

----

Can you tell if that error is still happens or maybe I should try with another data?
